### PR TITLE
Delete Charon V0 API

### PR DESCRIPTION
**Pull request by bschofield-va**
- set version 6.0.0-SNAPSHOT
- WebExceptionHandlerTest now uses fake controller
- test-request-file (and FileRpcRequestTest) now use V1 formatted request files
- omg... so much delete pain

### CODE REVIEW

File: charon/src/test/java/gov/va/api/lighthouse/charon/service/config/PathRewriteConfigTest.java [hunk 1]
```diff
@@ -31,9 +31,9 @@
   void pathIsRewritten() {
     assertThat(
         restTemplate.getForObject(
-            "http://localhost:" + port + "/charon/rpc/connections", VistalinkProperties.class));
+            "http://localhost:" + port + "/charon/v1/rpc/connections", VistalinkProperties.class));
     assertThat(
         restTemplate.getForObject(
-            "http://localhost:" + port + "/rpc/connections", VistalinkProperties.class));
+            "http://localhost:" + port + "/v1/rpc/connections", VistalinkProperties.class));
```
Comment (jhulbert-va) [inline charon/src/test/java/gov/va/api/lighthouse/charon/service/config/PathRewriteConfigTest.java:37]: Should this also go to `/charon/v1` for consistency with the one above?

File: charon-models/src/test/java/gov/va/api/lighthouse/charon/models/vprgetpatientdata/VprGetPatientDataAsserts.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import java.nio.charset.StandardCharsets;
+import lombok.SneakyThrows;
+
+public class VprGetPatientDataAsserts {
+
+  static final ObjectMapper XML_MAPPER =
+      new XmlMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+  @SneakyThrows
+  static <T> void assertDeserializedEquals(String xmlResponseResource, T expected) {
```
Comment (jhulbert-va) [inline charon-models/src/test/java/gov/va/api/lighthouse/charon/models/vprgetpatientdata/VprGetPatientDataAsserts.java:17]: Should we round trip this bad boi?

Comment (bschofield-va) [reply charon-models/src/test/java/gov/va/api/lighthouse/charon/models/vprgetpatientdata/VprGetPatientDataAsserts.java:17]: I don't think that is best here... we never _emit_ XLM representations so this is really about parsing the Vista response, not recreating.

File: charon/src/test/java/gov/va/api/lighthouse/charon/service/config/PathRewriteConfigTest.java (hunk 1 above)
Comment (bschofield-va) [reply charon/src/test/java/gov/va/api/lighthouse/charon/service/config/PathRewriteConfigTest.java:37]: no it's intentionally testing two paths to go to the same place

**Review by jhulbert-va — DISMISSED** [review]
LGTM

File: charon/requests/search-patient-coverage-request.json [hunk 3]
```diff
@@ -4,25 +4,21 @@
       "accessCode":"$VISTA_ACCESS_CODE",
       "verifyCode":"$VISTA_VERIFY_CODE"
    },
-   "target":{
-      "include":[
-         "host.docker.internal:18673:673:America/New_York"
-      ]
-   },
+  "vista"; "$VISTA",
```
Comment (jhulbert-va) [inline charon/requests/search-patient-coverage-request.json:7]: Ehrm...

File: charon/requests/read-patient-coverage-request.json [hunk 4]
```diff
@@ -1,19 +1,15 @@
...
-      "parameters":[
-         {
+  "principal":{
+    "applicationProxyUser":"LHS,APPLICATION PROXY",
+    "accessCode":"$VISTA_ACCESS_CODE",
+    "verifyCode":"$VISTA_VERIFY_CODE"
+  },
+  "vista":"$VISTA",
+  "rpc":{
+    "name":"LHS LIGHTHOUSE RPC GATEWAY",
```
Comment (jhulbert-va) [inline charon/requests/read-patient-coverage-request.json:9]: Should we rename this `lhs-rpc-gateway-gets.json`?

**Review by jhulbert-va — APPROVED** [review]

**Pull request merged**